adding casefile for DIVD case DIVD-2024-00039 (#846)
DIVD-NL/web-csirt: _cases/2024/DIVD-2024-00039.md

---
layout: case

> title: "Incorrect Authorization vulnerability in Apache OFBiz resulting in RCE"

Reviewer: ```suggestion
title: "Incorrect Authorization vulnerability in Apache OFBiz resulting in Remote Code Execution"
```

author: Wessel Baltus
lead: Wessel Baltus

> excerpt: "In Apache OFBiz, version 18.12.14and below, a Incorrect Authorization vulnerability exists that allows pre-authentication remote code execution (RCE) resulting in a attacker being able to execute arbitrary commands on the affected system by sending a specially crafted HTTP request."

Reviewer: ```suggestion
excerpt: "In Apache OFBiz, version 18.12.14 and below, an Incorrect Authorization vulnerability exists that allows pre-authentication remote code execution (RCE) resulting in an attacker being able to execute arbitrary commands on the affected system by sending a specially crafted HTTP request."
```

patch_status: Patch available
status : Open
start: 2024-09-29
timeline:
- start: 2024-09-29
  end:
  event: "DIVD starts researching the vulnerability."
- start: 2024-09-29
  end:
  event: "DIVD finds fingerprint, preparing to scan."
- start: 2024-09-29
  end:
  event: "Case opened, first version of this casefile"
- start: 2024-09-29
  end:
  event: "DIVD starts scanning the internet for vulnerable instances."
- start: 2024-10-01
  end:
  event: "DIVD starts notifying network owners with a vulnerable instance in their network"

---

## Summary

CVE-2024-38856 is a critical pre-authentication remote code execution (RCE) vulnerability in Apache OFBiz. The flaw stems from insufficient validation of the ProgramExport endpoint, which can be accessed without authentication. Attackers exploit this by chaining the ProgramExport endpoint with other publicly accessible endpoints, effectively bypassing authentication controls. This allows the execution of arbitrary code on vulnerable systems, leading to full system compromise. The vulnerability affects versions of OFBiz up to 18.12.14, and upgrading to version 18.12.15 is required to mitigate this threat.

## Recommendations

> The Apache OFBiz versions 18.12.14 and below are vulnerable. Upgrade to version 18.12.15 as soon as possible.

Reviewer: ```suggestion
The Apache OFBiz versions 18.12.14 and below are vulnerable. Upgrade to version 18.12.15 or higher as soon as possible.
```
